### PR TITLE
Add mechanism to edit views

### PR DESCRIPTION
**Pull request by gberaudo**

### CODE REVIEW

File: doc/administrator/editing.rst [hunk 1]
```diff
@@ -132,3 +132,13 @@ Example:
...
    * Only the geometry will be copied, the attributes will not be.
+
+Edit views
+----------
+
+To be able to edit PostgreSQL views a primary key must be manually configured.
+Add a layer metadata ``geotable_primary_key`` with value the name of the column to use as primary key.
+That column must be of type ``Integer``.
+
+Example:
```
Comment (sbrunner) [inline doc/administrator/editing.rst]: Shouldn't you add an extra new line to be a liste?

File: c2cgeoportal/lib/dbreflection.py [hunk 2]
```diff
@@ -161,16 +161,25 @@ def get_table(tablename, schema=None, session=None):
...
+                autoload=True,
+                autoload_with=engine,
+            )
+        else:
+            table = Table(
+                tablename, metadata,
+                schema=schema,
+                autoload=True,
+                autoload_with=engine,
+            )
```
Comment (sbrunner) [inline c2cgeoportal/lib/dbreflection.py]: easier to maintain: 
```
args = [tablename, metadata]
if primary_key is not None:
  # Ensure we have a primary key to be able to edit views
  args.append(Column(primary_key, Integer, primary_key=True))
table = Table(
  *args,
  schema=schema,
  autoload=True,
  autoload_with=engine,
)
```

File: c2cgeoportal/lib/dbreflection.py [hunk 3]
```diff
@@ -161,16 +161,20 @@ def get_table(tablename, schema=None, session=None):
...
+        if primary_key is not None:
+            # Ensure we have a primary key to be able to edit views
+            args.append(Column(primary_key, Integer, primary_key=True))
         table = Table(
-            tablename, metadata,
+            *args,
             schema=schema,
             autoload=True,
-            autoload_with=engine,
+            autoload_with=engine
```
Comment (sbrunner) [inline c2cgeoportal/lib/dbreflection.py]: ? for me it's better with a coma...

Comment (gberaudo) [reply c2cgeoportal/lib/dbreflection.py]: Strangely I get a 500 error with the coma.

Comment (gberaudo) [reply c2cgeoportal/lib/dbreflection.py]: What you wish is not supported in python prior to 3.6.
See https://stackoverflow.com/questions/33350454/syntaxerror-with-passing-kwargs-and-trailing-comma.

Comment (sbrunner) [reply c2cgeoportal/lib/dbreflection.py]: I never see that but it's true ...

**Pull request merged**